src/components/PlanRoom/RoomTables/Item.tsx

```diff
@@ -0,0 +1,184 @@
+import { useCallback, useState, SetStateAction } from 'react';
+import { Box, Textarea } from '@chakra-ui/react';
+import update from 'immutability-helper';
+import { useDrop, XYCoord, useDrag } from 'react-dnd';
+import ItemTypes from './ItemTypes';
+import { TypeItem, TableType } from './Types';
+import SaveBoard from '../SaveRoomTables/SaveBoard';
+
+let changeValueTable: Function;
+let changeValueTableBorderRadius: Function;
+let newTable: Function;
+let save: Function;
+const backgroundColor: string = '#211F1F';
+
+const Items = () => {
+  const [tables, setTables] = useState<{
+    [key: string]: { top: number; left: number; text: string; width: number; height: number; borderRadius: number };
+  }>({
+    0: { top: 0, left: 0, text: '0', width: 100, height: 50, borderRadius: 0 },
+  });
+
+  const moveTable = useCallback(
+    (id: string, left: number, top: number) => {
+      setTables(
+        update(tables, {
+          [id]: {
+            $merge: { left, top },
+          },
+        }),
+      );
+    },
+    [tables, setTables],
+  );
+
+  changeValueTable = useCallback(
+    (id: string, text: string) => {
+      setTables(
+        update(tables, {
+          [id]: {
+            $merge: { text },
+          },
+        }),
+      );
+    },
+    [tables, setTables],
+  );
+
+  changeValueTableBorderRadius = useCallback(
+    (id: string, borderRadius: number) => {
+      setTables(
+        update(tables, {
+          [id]: {
+            $merge: { borderRadius },
+          },
+        }),
+      );
+    },
+    [tables, setTables],
+  );
+
+  newTable = (id: number) => {
+    tables[id] = { top: 0, left: 0, text: id.toString(), width: 100, height: 50, borderRadius: 0 };
+    let top = Math.floor(Math.random() * 100) + 40;
+    if (top > 80) {
+      top = 80;
+    }
+    moveTable((id - 1).toString(), Math.floor(Math.random() * 800) + 50, top);
+  };
+
+  const [, drop] = useDrop(
+    () => ({
+      accept: ItemTypes.TABLE,
+      drop(item: TypeItem, monitor) {
+        const delta = monitor.getDifferenceFromInitialOffset() as XYCoord;
+        let left = item.left + delta.x;
+        let top = item.top + delta.y;
+        if (top > 319) {
+          const difference: number = top - 319;
+          top -= difference;
+        }
+        if (top < 0) {
+          top = 0;
+        }
+        if (left < 0) {
+          left = 0;
+        }
+        moveTable(item.id, left, top);
+        return undefined;
+      },
+    }),
+    [moveTable],
+  );
+
+  save = () => {
+    console.log(tables);
+    return tables;
```

Comment:
This function is pointless if it only returns data.

Reply:
this will be changed in further tasks